src/reporter/format-xspec-report.xsl

```diff
@@ -236,29 +241,29 @@
       </xsl:for-each>
     </tbody>
   </table>
-  <xsl:for-each select="x:scenario[not(@pending)]">
+  <xsl:for-each select="x:scenario[not(@pending) or .//x:test[parent::x:scenario][not(@pending)]]">

```

Comment:
This change looks correct to me. I was wondering, though, whether `x:test` can ever have a parent other than `x:scenario` in the foo-result.xml file. If you were just being defensive, that's fine.

Reply:
Yes, defensive. Parsing the result XML requires some defense, because XSpec may be used for testing XSpec itself. See the commit message of 8d23f5b9d1c6cda1777e47e97ac5338ac3378885. I don't remember exactly which tests are affected, but probably [generate-xspec-tests.xspec](https://github.com/xspec/xspec/blob/1b7e7824daa71d06294a51f25442e118a9bc4a68/test/generate-xspec-tests.xspec) is one (and only?) of such tests.
And that's also why I submitted #530.

Reply:
Thanks for explaining!